src/ast.zig

```diff
@@ -54,6 +55,32 @@ fn fullPtrTypeComponents(tree: Ast, info: full.PtrType.Components) full.PtrType
     return result;
 }
 
+/// Given an l_brace, find the corresponding r_brace.
+/// If no corresponding r_brace is found, return null.
+/// Useful for finding the extent of a block/scope if the syntax is valid.
+fn findMatchingRBrace(token_tags: []const TokenTag, l_brace: Ast.TokenIndex) ?Ast.TokenIndex {
+    std.debug.assert(token_tags[l_brace] == TokenTag.l_brace);
```

Comment:
You can avoid having to specify `TokenTag` if you instead do `token_tags[l_brace] == .l_brace`.
This is just nitpicking, so feel free to just ignore this comment. You said that you were still learning zig so I though I would let you know if you didn't know about this already.

Reply:
Ah good to know! Thank you for pointing it out.